tiny_dnn/core/backend.h

```diff
@@ -62,9 +62,12 @@ inline backend_t default_engine() {
 #ifdef CNN_USE_AVX
 #if defined(__AVX__) || defined(__AVX2__)
     return backend_t::avx;
+#else
+#error "your compiler does not support AVX"
 #endif
-#endif // CNN_USE_AVX
+#else
     return backend_t::internal;
+#endif
 }
```

Comment:
no comments for this. However, since we treat AVX as a backend do you think that we should enable it by default as it right now? I vote to not do it and use internal backend by default.